Example of FizzBuzz in kitten

bdoner opened the pull request:
An example of the FizzBuzz puzzle in Kitten.

evincarofautumn commented on examples/fizz-buzz.ktn:
```diff
@@ -0,0 +1,26 @@
+define buzzes(Int32 -> +IO +Fail):
+    -> n;
+    n 5 (%) 0 (=)
```
You can wrap these in `{ … } with (+Fail)` (or `do (with (+Fail)) { … }` if you prefer) to unsafely assert that the divisions won’t fail, and thereby remove the `+Fail` constraint on `buzzes` and `loop`.

There’s no real need for these operator calls to be postfix unless you’re taking advantage of composition—but by writing them in postfix you’ve revealed a good opportunity to factor out `(%) 0 (=)` into a definition called e.g. `divisible`.

It would be straightforward to remove the `n` variable as well (with `dup dup` to get a couple copies for the tests, and `drop` in the conditional branches where you ignore the value, e.g. `drop "Fizz"`) but that’s a matter of style and not terribly important.

evincarofautumn commented on examples/fizz-buzz.ktn:
```diff
@@ -0,0 +1,26 @@
+    n 5 (%) 0 (=)
+    n 3 (%) 0 (=)
+
+    match
```
Any particular reason you chose `match case true { … } case false { … }` over `if { … } else { … }`? (You can omit the `if` condition, same as `match`.)

evincarofautumn commented on examples/fizz-buzz.ktn:
```diff
@@ -0,0 +1,26 @@
+        case true:
+            "Buzz" say
+        case false:
+            n say
```
Since all branches end with `say`, you can factor it out to after the `match`/`if` if you change `n say` to `n show` so every branch returns a string. But moreover, you can remove the `+IO` constraint by moving the `say` out of `buzzes` and into `loop`, changing the return type of `buzzes` to `List<Char>`.

bdoner commented:
Changes pushed.

As you suggested I've moved the (%) 0 (=) into a `divisible` function which uses the `do (with (+Fail))` expression so I avoid having +Fail permission all the way back to the `loop` function.

I changed the match/case to if/else's it does make the intent more clear. At first I was focused on a different approach with pattern matching which I thought I needed match/case for. 

I have also removed the `say`'s and added a `show` in the `buzzes` function. This clearly makes more sense, but I am very focused on the language rather that being DRY 😄

evincarofautumn commented:
LGTM, committed in 64a00f3b8a01941152a26c06a8c1c731359f74f5. :)
